src/Param.h

```diff
@@ -42,6 +42,8 @@ class Param {
 
 	int doubleprecision = 0; // 0: float precision, 1: double precision
 
+	int engine = 1; // 1: Buttinger, 2: Kurganov, 3: KurganovATMP
```

Comment:
Add paper reference ?

Reply:
Ok I updated the description with a citing reference. Full reference is in the function documentation.